packages/ui/src/machines/authenticator/actors/resetPassword.ts

```diff
@@ -183,7 +183,7 @@ export function resetPasswordActor({ services }: ResetPasswordMachineOptions) {
       },
       services: {
         async resetPassword(context) {
-          const username = context.formValues?.username ?? context.username;
+          const { username } = context;
```

Comment:
I couldn't find anywhere why we couldn't use `context.username` here. In fact, the `confirmResetPassword`, the next step, uses the `context.username` so it has to be there otherwise everything would fail.

Reply:
Hmm, technically `context.formValues?.username` is where username values lives in initially.

It is then assigned to `context.username` as part of `setUsername` entry action in `resetPassword.submit`, so that's why `context.username` is always there for `resetPassword` and `confirmResetPassword` flow.

`setUsername` entry action:

https://github.com/aws-amplify/amplify-ui/blob/194ba9fff4ea20fd87d6b40d4c98b19f77bad727/packages/ui/src/machines/authenticator/actors/resetPassword.ts#L47-L49

Reply:
So that's the technical background why I think that check existed in the first place... but anyway as you wrote, it works just as well if we simply use 

```ts
const { username } = context;
```

Reply:
Ok that makes sense